cmd/deploy.go

```diff
@@ -7,62 +7,113 @@ import (
 	"github.com/boson-project/faas/docker"
 	"github.com/boson-project/faas/knative"
+	"github.com/boson-project/faas/progress"
 	"github.com/boson-project/faas/prompt"
 )
 
 func init() {
 	root.AddCommand(deployCmd)
 	deployCmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all configuration options - $FAAS_CONFIRM")
+	deployCmd.Flags().StringP("image", "i", "", "Optional full image name, in form [registry]/[namespace]/[name]:[tag] for example quay.io/myrepo/project.name:latest (overrides --repository) - $FAAS_IMAGE")
 	deployCmd.Flags().StringP("namespace", "n", "", "Override namespace into which the Function is deployed (on supported platforms).  Default is to use currently active underlying platform setting - $FAAS_NAMESPACE")
 	deployCmd.Flags().StringP("path", "p", cwd(), "Path to the function project directory - $FAAS_PATH")
+	deployCmd.Flags().StringP("repository", "r", "", "Repository for built images, ex 'docker.io/myuser' or just 'myuser'.  - $FAAS_REPOSITORY")
 }
 
 var deployCmd = &cobra.Command{
 	Use:   "deploy",
-	Short: "Deploy an existing Function project to a cluster",
-	Long: `Deploy an existing Function project to a cluster
+	Short: "Deploy or Update an existing Function project to a cluster",
+	Long: `Deploy or Update an existing Function project to a cluster
 
-Deploys the Function project in the current directory. A path to the project
-directory may be provided using the --path or -p flag. The image to be deployed
-must have already been created using the "build" command.
+Builds and Deploys the Function project in the current directory. A path to the project
+directory may be provided using the --path or -p flag. Reads the faas.yaml configuration file
+to determine the image name. An image and repository may be specified on the command line using the  --image or -i
+and --repository or -r flag.
+
+If the Function is already deployed, it is updated with a new container image that is pushed to a
+container image repository, and the Knative Service is updated.
 
 The namespace into which the project is deployed defaults to the value in the
 faas.yaml configuration file. If NAMESPACE is not set in the configuration,
 the namespace currently active in the Kubernetes configuration file will be
 used. The namespace may be specified on the command line using the --namespace
 or -n flag, and if so this will overwrite the value in the faas.yaml file.
+
+
 `,
 	SuggestFor: []string{"delpoy", "deplyo"},
-	PreRunE:    bindEnv("namespace", "path", "confirm"),
+	PreRunE:    bindEnv("image", "namespace", "path", "repository", "confirm"),
 	RunE:       runDeploy,
 }
 
 func runDeploy(cmd *cobra.Command, _ []string) (err error) {
-	config := newDeployConfig()
-	function, err := functionWithOverrides(config.Path, functionOverrides{Namespace: config.Namespace})
+
+	config := newDeployConfig().Prompt()
+
+	// Load Function from the config
+	// function, err := faas.NewFunction(config.Path)
+	// if err != nil {
+	// 	return
+	// }
```

Comment:
```suggestion
```

:wink:

Reply:
?? I have no idea why it is here, sorry.